packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/observe/useTransactionWatcher.ts

```diff
@@ -34,7 +32,7 @@ export type WatchedTransaction = {
   write?: Write;
   logs?: Log[];
   error?: BaseError;
-};
+} & ({ hash: Hex; writeId?: string } | { hash?: Hex; writeId: string } | { hash: Hex; writeId: string });
```

Comment:
prob don't need that last union?
```suggestion
} & ({ hash: Hex; writeId?: string } | { hash?: Hex; writeId: string });
```

Reply:
Yeah, I'd say it's not strictly accurate anymore though since it can be that both, `hash` and `writeId` are non-null.